python/pyarrow/_azurefs.pyx

```diff
@@ -107,7 +128,32 @@ cdef class AzureFileSystem(FileSystem):
         if account_key and sas_token:
             raise ValueError("Cannot specify both account_key and sas_token.")
 
-        if account_key:
+        if (tenant_id or client_id or client_secret):
+            if tenant_id and client_id and client_secret:
+                options.ConfigureClientSecretCredential(
+                    tobytes(tenant_id), tobytes(client_id), tobytes(client_secret)
+                )
+                self.tenant_id = tobytes(tenant_id)
+                self.client_id = tobytes(client_id)
+                self.client_secret = tobytes(client_secret)
+            elif client_id and not tenant_id and not client_secret:
+                options.ConfigureManagedIdentityCredential(tobytes(client_id))
+                self.client_id = tobytes(client_id)
+            elif not client_id:
+                raise ValueError(
+                    "client_id must be specified for both ClientSecretCredential and ManagedIdentityCredential. "
+                    "For ClientSecretCredential, provide tenant_id, client_id, and client_secret. "
+                    "For ManagedIdentityCredential, provide only client_id."
```

Comment:
I don't think we require that much verbosity (this is already specified on the API), let's maybe just say what is the problem?
`client_id must be specified if client_secret or tenant_id are provided`?